### PR TITLE
Code listing style fixes, remove extra line management

### LINKED ISSUE #662: Tutorial extra empty line bug
### Description

Extra line is render for code in tutorial.

### Checklist

- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/apple/swift-docc-render/issues).

### Expected Behavior

Show the correct highlighting for Line 25-29

image

image

```
{
  "content": [...],
  "fileName": "CustomizedSlothView.swift",
  "fileType": "swift",
  "highlights": [
    {
      "line": 25
    },
    {
      "line": 26
    },
    {
      "line": 27
    },
    {
      "line": 28
    },
    {
      "line": 29
    }
  ],
  "identifier": "01-creating-code-02-08.swift",
  "syntax": "swift",
  "type": "file"
}
```

### Actual behavior

Show the wrong highlighting (Due to extra empty line on Line 4 and Line 5).

There should be only 1 empty line (Line 3). Currently there will be 3 empty line(Line 3 - Line 5)

image

### Steps To Reproduce

The tutorial resource can be downloaded from https://developer.apple.com/documentation/xcode/slothcreator_building_docc_documentation_in_xcode

### Swift-DocC-Render Version Information

_No response_

### PR DESCRIPTION
**Pull request by dobromir-hristov**
closes #662 

rdar://109756906

## Summary

This PR refactors the css styling of the CodeListing in order to properly render new lines coming from code.

## Dependencies

- This builds upon #663
- probably fixes #502 properly

## Testing

This PR should be tested with as many examples as possible, across Tutorials and Doc pages.

We should test with scrollable content, multi-line content, and any combination of them.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [ ] ~Added tests~ - css mostly
- [ ] Ran `npm test`, and it succeeded
- [ ] Updated documentation if necessary

### CODE REVIEW

**Comment by dobromir-hristov** [comment]
@Kyle-Ye give this a try and let me know :)

**Comment by Kyle-Ye** [comment]
> @Kyle-Ye give this a try and let me know :)

Tried it. It seems to fix all the bug I know. Thanks

- [x] Missing new line in multi string swift code
- [x] Extra new line in tutorial
- [x] Proper new line copy behavior in tutorial.

File: src/utils/syntax-highlight.js [hunk 1]
```diff
@@ -192,7 +192,7 @@ function duplicateMultilineNode(element) {
 
   // wrap each new line with the current element's class
   const result = getLines(element.innerHTML)
-    .reduce((all, lineText) => `${all}<span class="${className}">${lineText || '\n\n'}</span>\n`, '');
+    .reduce((all, lineText) => `${all}<span class="${className}">${lineText || ''}</span>\n`, '');
```
Comment (Kyle-Ye) [inline src/utils/syntax-highlight.js:195]: Maybe we can remove this?
```suggestion
    .reduce((all, lineText) => `${all}<span class="${className}">${lineText}</span>\n`, '');
```

**Review by Kyle-Ye — APPROVED** [review]
LGTM

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Review by mportiz08 — APPROVED** [review]
Nice work!

Looks like this fixes both the tutorial and multiple line literal issue, and I also wasn't able to find any obvious regressions with any other complicated code listing we might render, like API Changes, collapsible REST code listings, etc

🚀

**Review by marinaaisa — APPROVED** [review]
Tested and seems to look good now 💯

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Pull request merged**